titus-common/src/main/java/com/netflix/titus/common/util/CollectionsExt.java

```diff
@@ -595,6 +595,31 @@ private CollectionsExt() {
         return -from - 1;
     }
 
+    /**
+     * Convert all keys to lower case, the behavior is undefined when values clash, and only one will be kept
+     */
+    public static <V> Map<String, V> toLowerCaseKeys(Map<String, V> entries) {
+        return entries.entrySet().stream()
```

Comment:
Direct for loop for better performance, as this code might be used in many places?

Reply:
Is there a unit test for these methods?

Reply:
the same performance argument could be made to any other of the `CollectionsExt` methods, I think this is fine, I will leave it as is for now to avoid early optimization.

Good point about the unit test, I will add one.